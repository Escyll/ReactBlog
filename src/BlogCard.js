import React from 'react';
import { observer } from 'mobx-react-lite';
import { Card, CardText, CardBody, CardImg, CardTitle, CardSubtitle } from 'reactstrap';
import classNames from "classnames";
import BlogPostImage from './BlogPostImage.jpg';

const BlogCard = observer(({ blogPost }) =>
    <Card
        className={classNames("BlogCard", {
            "border-2": blogPost.selected,
            "border-primary": blogPost.selected,
        })}>
        <CardImg top width="100%" src={BlogPostImage} alt={blogPost.title} />
        <CardBody>
            <CardTitle tag="h5">{blogPost.title}</CardTitle>
            <CardSubtitle tag="h6">By {blogPost.user.name}</CardSubtitle>
            <CardText style={{ marginTop: `20px` }}>By {blogPost.body}</CardText>
        </CardBody>
    </Card>
);

export default BlogCard;