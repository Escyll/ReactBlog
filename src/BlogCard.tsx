import React from 'react';
import { observer } from 'mobx-react-lite';
import { Card, CardText, CardBody, CardImg, CardTitle, CardSubtitle } from 'reactstrap';
import classNames from "classnames";
import BlogPostImage from './BlogPostImage.jpg';
import BlogPost from './BlogPost';

interface BlogCardProps {
    blogPost: BlogPost,
}

const BlogCard = observer(({ blogPost }: BlogCardProps) =>
    <Card
        className={classNames("BlogCard", {
            "border-2": blogPost.selected,
            "border-primary": blogPost.selected,
        })}>
        <CardImg top width="100%" src={BlogPostImage} alt={blogPost.title} />
        <CardBody>
            <CardTitle tag="h5">{blogPost.title}</CardTitle>
            <CardSubtitle tag="h6">By {blogPost.user.name}</CardSubtitle>
            <CardText style={{ marginTop: `20px` }}>{blogPost.body}</CardText>
        </CardBody>
    </Card>);

export default BlogCard;