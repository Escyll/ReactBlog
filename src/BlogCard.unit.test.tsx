import renderer from 'react-test-renderer';
import BlogCard from './BlogCard';

test('BlogCard', () => {
    const post = renderer.create(<BlogCard blogPost={{id: 1, title: "abc", body: "def", selected: false, userId: 1, user: {id: 1, name: "ghi"}}}/>);
    let tree = post.toJSON();
    expect(tree).toMatchSnapshot();
});

test('BlogCard selected', () => {
    const post = renderer.create(<BlogCard blogPost={{id: 1, title: "abc", body: "def", selected: true, userId: 1, user: {id: 1, name: "ghi"}}}/>);
    let tree = post.toJSON();
    expect(tree).toMatchSnapshot();
});