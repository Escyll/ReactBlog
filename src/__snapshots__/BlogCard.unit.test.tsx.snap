// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard 1`] = `
<div
  className="BlogCard card"
>
  <img
    alt="abc"
    className="card-img-top"
    src="BlogPostImage.jpg"
    width="100%"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      abc
    </h5>
    <h6
      className="card-subtitle"
    >
      By 
      ghi
    </h6>
    <p
      className="card-text"
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      def
    </p>
  </div>
</div>
`;

exports[`BlogCard selected 1`] = `
<div
  className="BlogCard border-2 border-primary card"
>
  <img
    alt="abc"
    className="card-img-top"
    src="BlogPostImage.jpg"
    width="100%"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      abc
    </h5>
    <h6
      className="card-subtitle"
    >
      By 
      ghi
    </h6>
    <p
      className="card-text"
      style={
        Object {
          "marginTop": "20px",
        }
      }
    >
      def
    </p>
  </div>
</div>
`;
